locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -0,0 +1,190 @@
+     *
+     * @param tcpPortRanges
+     */
+    public void setTcpPortRanges(List<String> tcpPortRanges) {
+        this.tcpPortRanges = BrooklynNetworkUtils.portRulesToRanges(tcpPortRanges);
+    }
+
+    public void setUdpPortRanges(ImmutableList<String> udpPortRanges) {
+        this.udpPortRanges = BrooklynNetworkUtils.portRulesToRanges(udpPortRanges);
+    }
+
+    /**
+     * Set to restict the range that the ports that are to be opened can be accessed from
+     *
+     * @param cidr
+     */
+    public void setCidr(String cidr) {
+        this.cidr = cidr;
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, Template template) {
+        super.customize(location, computeService, template);
+
+        inboundPorts = template.getOptions().getInboundPorts();
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (cidr != null) instance.setSshCidrSupplier(Suppliers.ofInstance(new Cidr(cidr)));
+
+        instance.customize(location, computeService, template);
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
+        super.customize(location, computeService, machine);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        ImmutableList.Builder<IpPermission> builder = ImmutableList.<IpPermission>builder();
+
+        builder.addAll(getIpPermissions(instance, tcpPortRanges, IpProtocol.TCP));
+
+        builder.addAll(getIpPermissions(instance, udpPortRanges, IpProtocol.UDP));
+
+        if (inboundPorts != null) {
+
+            for (int inboundPort : inboundPorts) {
+                IpPermission ipPermission = IpPermission.builder()
+                        .fromPort(inboundPort)
+                        .toPort(inboundPort)
+                        .ipProtocol(IpProtocol.TCP)
+                        .cidrBlock(instance.getBrooklynCidrBlock())
+                        .build();
+                builder.add(ipPermission);
+            }
+        }
+        instance.addPermissionsToLocation(machine, builder.build());
+    }
+
+    private List<IpPermission> getIpPermissions(JcloudsLocationSecurityGroupCustomizer instance, RangeSet<Integer> tcpPortRanges, IpProtocol protocol) {
+        List<IpPermission> ipPermissions = ImmutableList.<IpPermission>of();
+        if (tcpPortRanges != null) {
+             ipPermissions =
+                    FluentIterable
+                            .from(tcpPortRanges.asRanges())
+                            .transform(portRangeToPermission(instance, protocol))
+                            .toList();
+        }
+        return ipPermissions;
+    }
+
+    private Function<Range<Integer>, IpPermission> portRangeToPermission(final JcloudsLocationSecurityGroupCustomizer instance, final IpProtocol udp) {
```

Comment:
Can you rename the field `udp` to `protocol` as it could have the value TCP or UDP, so is a bit confusing.